apps/api/src/routes/applicationRouter.ts

```diff
@@ -836,63 +753,54 @@ applicationRouter.post(
 			try {
 				const applicationId = Number(request.params.applicationId);
 
-				const { userId } = request.session.user || {};
-				if (!userId) {
+				const user = request.session.user;
+				if (!user) {
 					response.status(401).json({ error: 'UNAUTHORIZED', message: 'User is not authenticated.' });
 					return;
 				}
 
-				const userMayEditResult = await validateUserPermissionForApplication({ userId, applicationId });
-				if (!userMayEditResult.success) {
-					switch (userMayEditResult.error) {
-						case 'SYSTEM_ERROR': {
-							response.status(500).json({ error: userMayEditResult.error, message: userMayEditResult.message });
-							return;
-						}
-						case 'NOT_FOUND': {
-							response.status(404).json({ error: userMayEditResult.error, message: userMayEditResult.message });
-							return;
-						}
-						case 'FORBIDDEN': {
-							response.status(403).json({ error: userMayEditResult.error, message: userMayEditResult.message });
-							return;
-						}
-						default: {
-							response.status(500).json({ error: 'SYSTEM_ERROR', message: userMayEditResult.message });
-							return;
-						}
-					}
+				const applicationResult = await getApplicationById({ applicationId });
+
+				if (!applicationResult.success) {
+					response.status(500).json({ error: 'SYSTEM_ERROR', message: 'Trouble' });
```

Comment:
Change the error message, this `applicationResult` object should contain the detailed message.
```suggestion
					response.status(500).json({ error: applicationResult.error, message: applicationResult.message });
```